### PR TITLE
Api 6811/evidence submission validation docs

### PR DESCRIPTION
**Pull request by nathanbwright**
## Description of change
Updates to the evidence submission docs.

## Original issue(s)
https://vajira.max.gov/browse/API-6811

### CODE REVIEW

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb [hunk 1]
```diff
@@ -471,7 +471,11 @@ class AppealsApi::V1::NoticeOfDisagreementsControllerSwagger
     operation :post, tags: NOD_TAG do
       key :operationId, 'postNoticeOfDisagreementEvidenceSubmission'
       key :summary, 'Get a location for subsequent evidence submission document upload PUT request'
-      key :description, ''
+      key :description, <<~DESC
+        This is the first step to submitting supporting evidence for an NOD.  (See the Evidence Uploads section above for additional information.)
+
+        The Notice of Disagreement GUID that is returned when the NOD is submitted, is supplied to this endpoint to ensure the NOD is in a valid state for sending supporting evidence documents.  Only NODs that selected the Evidence Submission lane are allowed to submit evidence documents up to 90 days after the NOD is received by VA.
+      DESC
       parameter name: 'uuid', 'in': 'path', required: true, description: 'Associated Notice of Disagreement UUID' do
```
Comment (ElleeB) [inline modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:479]: In the code we're expecting `params[:nod_id]`. Should this be included here? Forgive me, I still don't completely understand our Swagger documentation :/

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb [hunk 2]
```diff
@@ -508,14 +512,14 @@ class AppealsApi::V1::NoticeOfDisagreementsControllerSwagger
                 end
 
                 property :status do
-                  key :description, 'status description here...'
+                  key :description, ''
```
Comment (yakschuss) [inline modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:515]: Maybe remove the descriptions altogether if they're empty strings?

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb [hunk 3]
```diff
@@ -508,14 +512,14 @@ class AppealsApi::V1::NoticeOfDisagreementsControllerSwagger
...
                   key :type, :string
                   key :enum, %i[pending ...]
                   key :example, 'pending'
                 end
 
                 property :code do
-                  key :description, 'code description here...'
+                  key :description, ''
```
Comment (yakschuss) [inline modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:522]: Same comment

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb [hunk 4]
```diff
@@ -560,7 +656,7 @@ class AppealsApi::V1::NoticeOfDisagreementsControllerSwagger
     operation :put, tags: NOD_TAG do
       key :operationId, 'putNoticeOfDisagreementEvidenceSubmission'
       key :summary, 'Accepts Notice of Disagreement Evidence Submission document upload.'
-      key :description, 'Detailed description here...'
+      key :description, ''
```
Comment (yakschuss) [inline modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:659]: same

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb (hunk 1 above)
Comment (yakschuss) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:479]: yeah, it looks like the name of the parameter as documented is `uuid`, which should be changed to `nod_id`. Or perhaps we should change the controller attribute to be more consistent with how we refer to the UUID? 🤔

Comment (ElleeB) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:479]: I was thinking that, but I do feel that just calling it `nod_id` is so much more clear than `uuid`, even if it is less consistent.

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb (hunk 2 above)
Comment (ElleeB) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:515]: I'm seeing this in our NOD documentation as well. Is this a placeholder?

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb (hunk 3 above)
Comment (ElleeB) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:522]: Same same

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb (hunk 2 above)
Comment (ElleeB) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:515]: I'm seeing this in other places throughout the code...these are placeholders?

File: modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb (hunk 1 above)
Comment (nathanbwright) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:479]: Good points... What about `nod_uuid`?

Comment (ElleeB) [reply modules/appeals_api/app/swagger/appeals_api/v1/notice_of_disagreements_controller_swagger.rb:479]: There are points in the evidence submission controller where we are looking for evidence submissions via
`find_by(guid: params[:id])` and times when we are creating new submissions with `AppealsApi::EvidenceSubmissionRequestValidator.new(params[:nod_id]...` For me, the code is much more clear when it identifies that this is the `nod_id`, so for me it's about clarity. If this is only clear for me, then let's change it.

**Review by SlakrHakr — APPROVED** [review]

**Review by cilestin — APPROVED** [review]

**Pull request merged**